handler/spec.go

```diff
@@ -0,0 +1,23 @@
+package handler
+
+import "context"
+
+// Interface defines the building blocks of an operator's reconciliation logic.
+// Note there can be multiple Handlers reconciling the same object in a chain.
+// In that case they are guaranteed to be executed in order one after another.
+type Interface interface {
+	// EnsureCreated is called when the observed runtime object is created or
+	// updated. After the successful execution of EnsureCreated, systems being
+	// managed have created or updated system resources. This method must be
+	// idempotent.
+	EnsureCreated(ctx context.Context, obj interface{}) error
```

Comment:
How about this Request/Response thing we were talking about? For now it can have only `Obj` for me but let's get the interface right and as backward-compatible as we can.

```suggestion
	EnsureCreated(ctx context.Context, req Request) (Response, error)
```

Reply:
Check my latest commit. I added a suggestion but had to move the "name" functionality around because of import cycles. Note that I defined `Request` and `Response` as interface for the reason of ease of use within implementations. `return nil, nil` is cooler than `return handler.Response{}, nil` for instance. Does that work for you? I think it also makes the actually transported information immutable so that handlers cannot screw it up.

Reply:
We talked and we go with structs, but the response is a pointer.